c_src/brotli_nif.c

```diff
@@ -311,23 +313,36 @@ brotli_decoder_decompress_stream(ErlNifEnv *env, int argc,
     return BADARG;
   }
 
-  size_t available_out = 0;
   size_t available_in = input.size;
   const uint8_t *next_in = input.data;
+  size_t available_out = 0;
+  uint8_t *next_out = NULL;
 
-  BrotliDecoderResult result = BrotliDecoderDecompressStream(
-      *state, &available_in, &next_in, &available_out, NULL, NULL);
+  if (!enif_alloc_binary(0, &output)) {
+    return atom_error;
+  }
+
+  BrotliDecoderResult result = BROTLI_DECODER_RESULT_NEEDS_MORE_OUTPUT;
+  while (result == BROTLI_DECODER_RESULT_NEEDS_MORE_OUTPUT) {
+    result = BrotliDecoderDecompressStream(*state,
+                                           &available_in, &next_in,
+                                           &available_out, &next_out, NULL);
+    size_t output_chunk_size = 0; // Request all available output.
+    const uint8_t* data = BrotliDecoderTakeOutput(*state, &output_chunk_size);
+    if (output_chunk_size && result != BROTLI_DECODER_RESULT_ERROR) {
+      enif_realloc_binary(&output, output_chunk_size + output.size);
+      memcpy(output.data + output.size - output_chunk_size, data, output_chunk_size);
+    }
+  }
 
   switch (result) {
   case BROTLI_DECODER_RESULT_SUCCESS:
-  case BROTLI_DECODER_RESULT_NEEDS_MORE_OUTPUT:
-    return atom_ok;
-
   case BROTLI_DECODER_RESULT_NEEDS_MORE_INPUT:
-    return atom_more;
-
+    return enif_make_tuple2(env, atom_ok, enif_make_binary(env, &output));
```

Comment:
I think it is possible memory leak place because you are not release output here.

Reply:
No - it will be garbage collected.